File: QuickSort/quicksort.java
import java.util.*;

class quicksorting {
    List<Integer> Arr;

    public quicksorting(List<Integer> Arr) {
        this.Arr = Arr;
    }

    public void swap(int i, int j) {
        int temp = Arr.get(i);
        Arr.set(i, Arr.get(j));
        Arr.set(j, temp);
    }

    public int partition(int start, int end) {
        int i = start - 1;
        int pivot = Arr.get(end);
        for (int j = start; j < end; j++) {
            if (Arr.get(j) <= pivot) {
                i++;
                swap(i, j);
            }
        }
        swap(i + 1, end);

        return i + 1;
    }

    public void quicksort(int start, int end) {
        if (start < end) {
            int q = partition(start, end);
            quicksort(start, q - 1);
            quicksort(q + 1, end);
        }

    }

    public void printArr() {
        System.out.print("Array after quicksorting: ");
        System.out.print(Arr);
        System.out.println();
    }
}

class quicksort {
    public static void main(String[] args) {

        List<Integer> Arr = new ArrayList<>();

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the Array to sort: ");
        String st = sc.nextLine();
        for (String i : st.split(" ")) {
            Arr.add(Integer.parseInt(i));
        }

        quicksorting qsort = new quicksorting(Arr);

        qsort.quicksort(0, Arr.size() - 1);

        qsort.printArr();

        sc.close();

    }
}

File: QuickSort/quicksort.py




class QuickSort:
    Arr=[]
    def __init__(self,Arr):
        self.Arr=Arr
        
    def swap(self,i,j):
        Arr=self.Arr
        temp=Arr[i]
        Arr[i]=Arr[j]
        Arr[j]=temp
        self.Arr=Arr
        
    def partition(self,start,end):
        Arr=self.Arr
        pivot=Arr[end]
        i=start-1
        for j in range(start,end):
            if Arr[j] < pivot:
                i+=1
                self.swap(i,j)
        self.swap(i+1,end)
        return i+1
    
    def quicksort(self,start,end):
        if start<end :
            q=self.partition(start,end)
            self.quicksort(start,q-1)
            self.quicksort(q+1,end)
    
    def printArr(self):
        print('printing Arr after quick sort: ',end='')
        for i in self.Arr:print(i,end=' ')
        print()
    
class Main:
    def main():
        print("Enter the list to sort: ",end='')
        Arr=[int(i) for i in input().split()]
        qsort=QuickSort(Arr)
        qsort.quicksort(0,len(Arr)-1)
        qsort.printArr()
        
    if __name__=="__main__":
        main()
        
        
